src/Servers/Kestrel/Core/src/Internal/Http/IHttpHeadersHandler.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+
+public interface IHttpHeadersHandler
+{
+    void OnStaticIndexedHeader(int index);
+    void OnStaticIndexedHeader(int index, ReadOnlySpan<byte> value);
```

Comment:
These probably don't need to be on the public interface. You only put them here to minimize breaks, right?

Reply:
While true, it looks like we've already shipped it starting in .NET 5. I don't see a reason to take another breaking change right now. If we do decide to break it and keep only what we need, we should probably send it through API review and do it all at once. We don't need the `bool endStream` for HTTP/1.x either for example.

Reply:
Yeah, I left it unchanged to avoid breaks.